omron_position_controller:
  cmd_vel_topic: {
    type: string,
    default_value: "/cmd_vel",
    description: "Topic to read when not in chain",
    validation: {
      not_empty<>: null
    }
  }
  cmd_pos_topic: {
    type: string,
    default_value: "/cmd_pos",
    description: "Topic to read when not in chain",
    validation: {
      not_empty<>: null
    }
  }
  feedback:
    k1: {
      type: double,
      default_value: 0.0,
      description: "k1 gain of the controller. pp 1242-1243 Springer handbook of Robotics"
    }
    k2: {
      type: double,
      default_value: 0.0,
      description: "k2 gain of the controller"
    }
    k3: {
      type: double,
      default_value: 0.0,
      description: "k3 gain of the controller"
    }
#  tf:
#    use_tf: {
#      type: bool,
#      default_value: true,
#      description: "broadcast pose through tf"
#    }
#    from: {
#      type: string,
#      default_value: "",
#      description: "transform from ...",
#      validation: {
#        not_empty<>: null
#      }
#    }
#    to: {
#      type: string,
#      default_value: "",
#      description: "transform to ...",
#      validation: {
#        not_empty<>: null
#      }
#    }
  interfaces:
    velocity:
      prefix: {
        type: string,
        description: "Prefix of the interfaces"
      }
      command: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
      states: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
    pose:
      prefix: {
        type: string,
        default_value: "omron_base/pose",
        description: "Prefix of the interfaces"
      }
      states: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }
    reference:
      prefix: {
        type: string,
        description: "Prefix of the interfaces"
      }
      command: {
        type: string_array,
        default_value: [],
        description: "",
        validation: {
          unique<>: null
        }
      }

